core/src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/workspacemanager/HttpWorkspaceManagerDAOSpec.scala

```diff
@@ -319,4 +319,59 @@ class HttpWorkspaceManagerDAOSpec
                                                     ArgumentMatchers.eq("test_job_id")
     )
   }
+
+  behavior of "listWorkspaces"
+
+  it should "should only make a single request when no workspaces are returned" in {
+    val workspaceApi = mock[WorkspaceApi]
+    val responseList = new WorkspaceDescriptionList().workspaces(List().asJava)
+    when(workspaceApi.listWorkspaces(ArgumentMatchers.eq(0), anyInt(), any())).thenReturn(responseList)
+    val wsmDao = new HttpWorkspaceManagerDAO(getApiClientProvider(workspaceApi = workspaceApi))
+    wsmDao.listWorkspaces(testContext) shouldBe empty
+    verify(workspaceApi).listWorkspaces(ArgumentMatchers.eq(0), anyInt(), any())
+  }
+
+  it should "should only make one requests when fewer than workspaces the request batch size of are returned" in {
+    val workspaceApi = mock[WorkspaceApi]
+    val ws = new WorkspaceDescription()
+    when(workspaceApi.listWorkspaces(ArgumentMatchers.eq(0), ArgumentMatchers.eq(2), any()))
+      .thenReturn(new WorkspaceDescriptionList().workspaces(List(ws).asJava))
+    val wsmDao = new HttpWorkspaceManagerDAO(getApiClientProvider(workspaceApi = workspaceApi))
+    wsmDao.listWorkspaces(testContext, 2) should have length 1
+    verify(workspaceApi).listWorkspaces(ArgumentMatchers.eq(0), ArgumentMatchers.eq(2), any())
```

Comment:
I don't see how this verifies that the mock only got called once.

Reply:
the default verification behavior is to verify once